### ISSUE TITLE
OptionButton is not usable as the popup menu closes instantly

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#42646]
**Godot version:**
`master` (28e45ec)
(Not reproducible in `3.2`, related to Window/Popup changes.)

**OS/device including version:**
Mageia 8 x86_64
Plasma, KWin on X11

**Issue description:**
When using an OptionButton in the editor on X11, the popup menu with the list of options closes instantly, so no value can be selected.

Additionally, it may appear as the wrong location:
![Screenshot_20201008_170454](https://user-images.githubusercontent.com/4701338/95477316-7fd80d80-0988-11eb-98ff-d7a9d3f8cc6e.png)
(here I clicked the "+" button on the "ui_accept" line)

**Steps to reproduce:**
- Find an editor plugin which uses OptionButtons, e.g. the InputMap editor
- Open the menu by clicking, it should close instantly

**Minimal reproduction project:**
n/a

**Comment by akien-mga** [comment]
Another example is the "Create" plugin for Plugins:
![Screenshot_20201008_170757](https://user-images.githubusercontent.com/4701338/95477642-db0a0000-0988-11eb-93e4-aa860df3772c.png)

Note that the OptionButton popup is not misplaced here, but it autocloses. On the other hand, the Create Plugin popup itself is in a weird location, looks like it calculates its offset based on the ProjectSettings, but then it's applied from the top-left of the screen, instead of the parent window. (Might be fixed by #42117?)

**Comment by bruvzg** [comment]
> Additionally, it may appear as the wrong location

Same on macOS, this part is not X11 specific.

On macOS `OptionBox` popup is usable as long as mouse cursor is on top of it or it's parent window, if mouse moves over any other Godot window (but not other app windows) popup closes.

All popup also print `window_get_current_screen` error on opening for the first time: 
```
ERROR: Condition "!windows.has(p_window)" is true. Returning: -1
   at: window_get_current_screen (platform/osx/display_server_osx.mm:2492)
```

**Comment by bruvzg** [comment]
Wrong popup locations should be fixed by #37506, but it needs rebase.

**Comment by pouleyKetchoupp** [comment]
On Windows:
I can confirm both the instant close and the misplacement.

On Linux/Ubuntu:
I can reproduce both too but the add button menu instantly closes only after I've pressed the edit button to modify a key first.

@bruvzg Yes some changes from #37506 might help, but (as you mentioned) it needs rebase and it will probably require some rework because both the X11 display server and PopupMenu have been modified to solve similar issues.

**Comment by Chaosus** [comment]
@akien-mga A duplicate of #40227?

**Comment by akien-mga** [comment]
Yes.
